fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanAccountDomainServiceJpa.java

```diff
@@ -150,6 +155,13 @@ public LoanTransaction makeRepayment(final Loan loan, final CommandProcessingRes
             final LocalDate transactionDate, final BigDecimal transactionAmount, final PaymentDetail paymentDetail, final String noteText,
             final String txnExternalId, final boolean isRecoveryRepayment, boolean isAccountTransfer, HolidayDetailDTO holidayDetailDto,
             Boolean isHolidayValidationDone, final boolean isLoanToLoanTransfer) {
+
+        final List<ApiParameterError> dataValidationErrors = new ArrayList<>();
+        final DataValidatorBuilder baseDataValidator = new DataValidatorBuilder(dataValidationErrors).resource("loan.transaction");
+        final GlobalConfigurationPropertyData configuration = this.configurationReadPlatformService
+                .retrieveGlobalConfiguration("Enable-Loan-Overpayment");
```

Comment:
To make the current behaviour default, would it not make sense to have a configuration value that works the other way around - i.e. call it "block-loan-overpayment", and if it is present & enabled then overpayments are blocked.

Reply:
thanks @ptuomola i did not think of this suggested approach. let me try just that